test/autoCompletionFix.test.ts

```diff
@@ -50,6 +50,28 @@ describe('Auto Completion Fix Tests', () => {
     });
   }
 
+  /**
+   * Generates a completion list for the given document and caret (cursor) position.
+   * @param content The content of the document.
+   * Alternatively, `position` can be omitted if the caret is located in the content using `|` bookends.
+   * For example, `content = 'ab|c|d'` places the caret over the `'c'`, at `position = 2`
+   * @returns A list of valid completions.
+   */
+  function parseCaret(content: string): Promise<CompletionList> {
+    // console.log(`was: len: ${content.length}, content: "${content}"`);
+    const position = content.search(/\|[^]\|/); // | -> any char including newline -> |
+    content = content.substring(0, position) + content.substring(position + 1, position + 2) + content.substring(position + 3);
```

Comment:
How about throwing an error if the caret is not found?

Reply:
Done